tempto-core/src/main/java/com/teradata/tempto/hive/HivePlugin.java

```diff
@@ -0,0 +1,43 @@
+package com.teradata.tempto.hive;
+
+import com.teradata.tempto.fulfillment.table.TableDefinition;
+import com.teradata.tempto.fulfillment.table.TableHandle;
+import com.teradata.tempto.fulfillment.table.hive.HiveDataSource;
+import com.teradata.tempto.internal.convention.tabledefinitions.FileBasedHiveDataSource;
+import com.teradata.tempto.spi.Plugin;
+import com.teradata.tempto.spi.convention.tabledefinitions.ConventionTableDefinitionDescriptor;
+
+import java.util.Optional;
+
+import static com.teradata.tempto.fulfillment.table.TableHandle.tableHandle;
+import static com.teradata.tempto.fulfillment.table.hive.HiveTableDefinition.hiveTableDefinition;
+
+public class HivePlugin
+    implements Plugin
+{
+    @Override
+    public TableDefinition getConventionTableDefinition(ConventionTableDefinitionDescriptor descriptor)
```

Comment:
I would rather have `Plugin` expose sth like `ConventionTableDefinitionProvider`.
And then provider would have method `getConventionTableDefinition`. 
We do not want to pass `Plugin` interface all around the codebase to places when only subset of things exposed by it are needed.